purchases/src/main/kotlin/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -36,7 +36,7 @@ import java.net.URLConnection
 import java.util.Date
 import kotlin.time.Duration
 
-class HTTPClient(
+internal class HTTPClient(
```

Comment:
making these internal is breaking the compilation because AmazonBackend is public and it exposes these (which are internal). Not sure if it's better to break semver and make AmazonBackend internal or expose these.  cc: @tonidero

Reply:
Huh... I would have expected that since `AmazonBackend` which is public in the `feature/amazon` module, has an internal class in the constructor `BackendHelper` (note that the amazon module is not exposing the common module), Kotlin should have given an error, that would force us to make the constructor of that class internal at least...

Not sure why that's not the case... currently, if I try to create an `AmazonBackend` in the API tests, I can't really do it...

So I guess they would be able to create one if someone also included the common module. And since we are making these internal, that would indeed be a breaking change... However, honestly, no-one should be creating these classes directly, so I'm kinda leaning towards making them internal